packages/editor/src/components/media-placeholder/index.js

```diff
@@ -158,12 +158,26 @@ class MediaPlaceholder extends Component {
 
 		const allowedTypes = this.getAllowedTypes();
 
+		let instructions = labels.instructions || '';
+		if ( ! instructions ) {
+			instructions = __( 'Drag a media file, upload a new one or select a file from your library.' );
+
+			if ( isEmpty( difference( allowedTypes, [ 'video' ] ) ) ) {
+				instructions = __( 'Drag a video, upload a new one or select a file from your library.' );
+			} else if ( isEmpty( difference( allowedTypes, [ 'image' ] ) ) ) {
+				instructions = __( 'Drag an image, upload a new one or select a file from your library.' );
+			} else if ( isEmpty( difference( allowedTypes, [ 'audio' ] ) ) ) {
+				instructions = __( 'Drag an audio, upload a new one or select a file from your library.' );
+			} else if ( isEmpty( difference( allowedTypes, [ 'image', 'video' ] ) ) ) {
+				instructions = __( 'Drag an image or a video, upload a new one or select a file from your library.' );
+			}
```

Comment:
Instead of adding all these if/else here, can we use just a generic string here and pass these as props where appropriate? that way we stay consistent in how we deal with labels in this component.

Reply:
I also wonder if we should have a generic "Media" in the text if the `allowedTypes` contains more than one item.

Reply:
This is already the case in https://github.com/WordPress/gutenberg/pull/10861/files#diff-56d6af193779467f7a4cb10e85c26ef3R170, unless I'm missing something.